### PR TITLE
[EOC-75] clean up lists reducer

### PR DESCRIPTION
**Pull request by alemikolo**
Commits to review: Remove shopping from names, Use composition for lists and items reducers, Change product to item in action names

### CODE REVIEW

File: client/src/modules/shopping-list/components/InputBar/model/actionTypes.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-  ADD_PRODUCT_REQUEST: 'product/ADD_PRODUCT_REQUEST',
-  ADD_PRODUCT_SUCCESS: 'product/ADD_PRODUCT_SUCCESS',
-  TOGGLE_PRODUCT_FAILURE: 'product/TOGGLE_PRODUCT_FAILURE',
-  TOGGLE_PRODUCT_REQUEST: 'product/TOGGLE_PRODUCT_REQUEST',
-  TOGGLE_PRODUCT_SUCCESS: 'product/TOGGLE_PRODUCT_SUCCESS',
-  VOTE_FOR_PRODUCT_FAILURE: 'product/VOTE_FOR_PRODUCT_FAILURE',
-  VOTE_FOR_PRODUCT_REQUEST: 'product/VOTE_FOR_PRODUCT_REQUEST',
-  VOTE_FOR_PRODUCT_SUCCESS: 'product/VOTE_FOR_PRODUCT_SUCCESS'
+export const ItemActionTypes = Object.freeze({
+  ADD_ITEM_FAILURE: 'item/ADD_ITEM_FAILURE',
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/components/InputBar/model/actionTypes.js]: do we need `_ITEM_` part?

File: client/src/modules/shopping-list/model/actionTypes.js [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
-  RESTORE_REQUEST: 'shopping-list/RESTORE_REQUEST',
-  RESTORE_SUCCESS: 'shopping-list/RESTORE_SUCCESS',
-  UPDATE_FAILURE: 'shopping-list/UPDATE_FAILURE',
-  UPDATE_REQUEST: 'shopping-list/UPDATE_REQUEST',
-  UPDATE_SUCCESS: 'shopping-list/UPDATE_SUCCESS'
+export const ListActionTypes = Object.freeze({
+  ARCHIVE_FAILURE: 'list/ARCHIVE_FAILURE',
+  ARCHIVE_REQUEST: 'list/ARCHIVE_REQUEST',
+  ARCHIVE_SUCCESS: 'list/ARCHIVE_SUCCESS',
+  CREATE_LIST_FAILURE: 'list/CREATE_LIST_FAILURE',
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/actionTypes.js]: do we need '_LIST_' here?

File: client/src/modules/shopping-list/model/reducer.js [hunk 3]
```diff
@@ -1,23 +1,59 @@
...
-    case ShoppingListActionTypes.FETCH_META_DATA_SUCCESS:
+    case ItemActionTypes.ADD_ITEM_SUCCESS:
+      return {
+        ...state,
+        products: [...state.products, action.payload.product]
+      };
+    case ItemActionTypes.TOGGLE_ITEM_SUCCESS:
+      return {
+        ...state,
+        products: [...state.products].map(product =>
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/reducer.js]: `map` creates new array - spread is not needed here

File: client/src/modules/shopping-list/model/reducer.js [hunk 4]
```diff
@@ -1,23 +1,59 @@
...
+                ...action.payload.product,
+                isOrdered: action.payload.product.isOrdered
+              }
+            : product
+        )
+      };
+    case ItemActionTypes.VOTE_FOR_ITEM_SUCCESS:
+      return {
+        ...state,
+        products: [...state.products].map(product =>
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/reducer.js]: same here with `map`

File: server/controllers/shoppingList.js [hunk 5]
```diff
@@ -2,22 +2,35 @@ const ShoppingList = require('../models/shoppingList.model');
...
-      ? resp.status(400).send({ message: err.message })
-      : resp
-          .location(`/shopping-list/${doc._id}`)
-          .status(201)
-          .send(doc);
+    if (err) {
+      return resp.status(404).send({ message: err.message });
+    }
+
+    if (!doc) {
```
Comment (marekrozmus) [inline server/controllers/shoppingList.js]: not needed for `save` - error should already handle such situations

File: client/src/modules/shopping-list/model/reducer.js [hunk 6]
```diff
@@ -22,10 +22,10 @@ const items = (state, action) => {
             : product
         )
       };
-    case ItemActionTypes.VOTE_FOR_ITEM_SUCCESS:
+    case ItemActionTypes.VOTE_FOR_SUCCESS:
       return {
         ...state,
-        products: [...state.products].map(product =>
+        products: state.products.map(product =>
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/model/reducer.js]: why did you removed spread from here? I remember that there was some problem that react didn't saw changes in state and was not re-rendering the app. Did you test it??

File: client/src/modules/shopping-list/model/reducer.js [hunk 7]
```diff
@@ -80,19 +80,19 @@ const lists = (state = {}, action) => {
         [action.payload.listId]: action.payload.data
       };
     }
-    case ItemActionTypes.ADD_ITEM_SUCCESS: {
+    case ItemActionTypes.ADD_SUCCESS: {
       return {
         ...state,
         [action.payload.listId]: items(state[action.payload.listId], action)
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/model/reducer.js]: Create const currentList = state[action.payload.listId]  before return ,
and pass this const to items reducer 
In every case

File: client/src/modules/shopping-list/model/reducer.js (hunk 6 above)
Comment (alemikolo) [reply client/src/modules/shopping-list/model/reducer.js]: Map returns new array, I tested it, everything works good

**Pull request merged**